lib/src/session.dart

```diff
@@ -36,12 +100,186 @@ class Session with ChangeNotifier {
+        data: htmlData, baseUrl: WebUri("https://app.talkjs.com"));
+  }
+
+  void _onLoadStop(InAppWebViewController controller, WebUri? url) async {
+    if (kDebugMode) {
+      print('📗 session._onLoadStop ($url)');
+    }
+
+    if (_webViewController == null) {
+      _webViewController = controller;
+
+      // Wait for TalkJS to be ready
+      final js = 'await Talk.ready;';
+
+      if (kDebugMode) {
+        print('📗 session callAsyncJavaScript: $js');
+      }
+
+      await controller.callAsyncJavaScript(functionBody: js);
+
+      _execute('let futures = []');
+
+      // If the `me` property has already been initialized, then create the user and the session
+      if ((_me != null) && (!_sessionInitialized)) {
+        _execute('const me = new Talk.User(${me.getJsonString()});');
+        createSession(
+          execute: _execute,
+          session: this,
+          variableName: 'me',
+        );
+
+        if (onMessage != null) {
+          _execute(
+              'session.onMessage((event) => window.flutter_inappwebview.callHandler("JSCOnMessage", JSON.stringify(event)));');
+        }
+
+        if ((unreads != null) && (unreads!.onChange != null)) {
+          _execute(
+              'session.unreads.onChange((event) => window.flutter_inappwebview.callHandler("JSCOnUnreadsChange", JSON.stringify(event)));');
+        }
+
+        if (_enablePushNotifications != null) {
+          await _setOrUnsetPushRegistration();
+        }
+
+        _sessionInitialized = true;
+        _completer.complete();
+      }
+    }
+  }
+
+  Future<dynamic> _execute(String statement) {
+    if (kDebugMode) {
+      print('📗 session.execute: $statement');
+    }
+
+    // We're sure that _execute only gets called with a valid _webViewController
+    return _webViewController!.evaluateJavascript(source: statement);
+  }
 
-  Session(
-      {required this.appId,
-      this.signature,
-      this.enablePushNotifications = false});
+  Future<dynamic> _executeAsync(String statement) async {
+    if (kDebugMode) {
+      print('📗 session.executeAsync: $statement');
+    }
+
+    // We're sure that _execute only gets called with a valid _webViewController
+    final result =
+        await _webViewController!.callAsyncJavaScript(functionBody: statement);
+    return result?.value;
+  }
+
+  void _jscOnMessage(List<dynamic> arguments) {
+    final message = arguments[0];
+
+    if (kDebugMode) {
+      print('📗 session._jscOnMessage: $message');
+    }
+
+    onMessage?.call(Message.fromJson(json.decode(message)));
+  }
+
+  void _jscOnUnreadsChange(List<dynamic> arguments) {
+    final List<dynamic> unreadsJson = json.decode(arguments[0]);
+
+    if (kDebugMode) {
+      print('📗 session._jscOnUnreadsChange: $unreadsJson');
+    }
+
+    unreads?.onChange?.call(
+      unreadsJson.map((unread) => UnreadConversation.fromJson(unread)).toList(),
+    );
+  }
+
+  Future<dynamic> _setOrUnsetPushRegistration() {
+    List<String> statements = [];
```

Comment:
This should be a single `String` since there's no scenario where the fcmToken will be available as well as the apnsToken being available. 

So in the checks below we should do:

```dart
if (fcmToken != null) {
    
} else if (apnsToken != null {

}
```

Reply:
👍